### PR TITLE
Game List Sort and Filters

### PR DESCRIPTION
**Pull request by devo1929**
This provides options for a few things:

- sort alphabetically
- show friend games only
- hide locked games
- hide passworded games
- hide incompatible games
- max player count

It creates a new panel that appears on top of the game list. This panel is where you'll find controls to modify the filters above.

There are two buttons added above the Game List as well. These buttons are a toggle button to quickly enable alphabetical sort and a button to quickly open the game filters panel on top of the game list. This filters toggle button will appear toggled "on" when there are filters enabled that are not the defaults.

All filter settings are stored to the settings ini (ra2md.ini).

The icons required for these new toggle buttons are located in the cncnet-yr-client-package:
https://github.com/CnCNet/cncnet-yr-client-package/pull/46

### CODE REVIEW

**Comment by Metadorius** [comment]
Some screenshots would definitely help.

**Comment by devo1929** [comment]
Screenshot 2021-07-04 143857

Screenshot 2021-07-04 143931

**Comment by Rampastring** [comment]
I think this new tab should be next to the CnCNet tab as it's still strictly CnCNet-related.

**Comment by Metadorius** [comment]
I think this is the case where a new control wouldn't hurt, perhaps something like a dropdown with other controls?

**Comment by Rampastring** [comment]
That's actually a great idea, then there wouldn't be a need for it to reside as an extra tab in the options menu. Basically a button that shows any XNAControl instance below itself when clicked on should be easy to implement and a good addition to the `XNAUI` library.

**Comment by devo1929** [comment]
I'd be willing to build something like that. I don't know if I have access to the XNAUI repo, currently, but I can make it in cncnet client for now.

**Comment by Metadorius** [comment]
> I'd be willing to build something like that. I don't know if I have access to the XNAUI repo, currently, but I can make it in cncnet client for now.

It's in Rampastring's repos.

**Comment by devo1929** [comment]
I'm typically not a fan of using drop downs or popups for more complex UIs. If it were simply a multi select, then I think that type of UI would work for this (the max count filter ruins that). 

What if it were a child panel of the GameListBox itself? IE, clicking on the new filter button would "swap" the list view for a filter options view. This filter options view would then have the Save and Cancel buttons as it currently does, which would then toggle back to the game list.

**Comment by devo1929** [comment]
I made some changes. Instead of the options window route, clicking the Filters button will show a panel that sits on top of the game list. The contents look the same as the screen shot above when it was in the options window, except that the sort alpha checkbox was removed, because it's quicker to just use the sort alpha button that's right above this panel.
Screenshot:

Screenshot 2021-07-06 125121

**Comment by alexp8** [comment]
looks great, please approve and merge @Metadorius

File: DTAConfig/OptionsWindow.cs [hunk 1]
```diff
@@ -68,6 +68,7 @@ public override void Initialize()
             var updaterOptionsPanel = new UpdaterOptionsPanel(WindowManager, UserINISettings.Instance);
             updaterOptionsPanel.OnForceUpdate += (s, e) => { Disable(); OnForceUpdate?.Invoke(this, EventArgs.Empty); };
 
+
```
Comment (Metadorius) [inline DTAConfig/OptionsWindow.cs:71]:
```suggestion
```

**Review by Metadorius — APPROVED** [review]
The code quality is fine and the implementation is pretty good.

**Pull request merged**